### ISSUE TITLE
BUG: In Windows, java.lang.NoSuchMethodError occurred while unzipping clojure-tools.zip

### ISSUE THREAD

**Issue by sohang3112** [BetterThanTomorrow/calva#2508]
## Steps to Reproduce
* Ran *Calva: Getting Started REPL* in VS Code
* `java.lang.NoSuchMethodError` error occurred during Jack-In while unzipping `clojure-tools.zip`

## Error Traceback
```
️ Starting the REPL ⚡️ using the below command line:
pushd c:\Users\SOHANG~1\AppData\Local\Temp\betterthantomorrow.calva\h0e5cl & java -jar ".calva\deps.clj.jar" -Sdeps "{:deps {nrepl/nrepl {:mvn/version,""1.0.0""},cider/cider-nrepl {:mvn/version,""0.28.5""}}}" -M -m nrepl.cmdline --middleware "[cider.nrepl/cider-middleware]" & popd
Clojure tools not yet in expected location: C:\Users\SohangChopra\.deps.clj\1.11.2.1446\ClojureTools\clojure-tools-1.11.2.1446.jar
Downloading https://github.com/clojure/brew-install/releases/download/1.11.2.1446/clojure-tools.zip to C:\Users\SohangChopra\.deps.clj\1.11.2.1446\ClojureTools\clojure-tools.zip
Unzipping C:\Users\SohangChopra\.deps.clj\1.11.2.1446\ClojureTools\clojure-tools.zip ...
Exception in thread "main" 
java.lang.NoSuchMethodError: java.util.zip.Checksum.update([B)V
        at borkdude.deps$unzip.invokeStatic(deps.clj:429)
        at borkdude.deps$clojure_tools_install_BANG_.invokeStatic(deps.clj:597)
        at borkdude.deps$_main$fn__229.invoke(deps.clj:924)
        at borkdude.deps$_main.invokeStatic(deps.clj:924)
        at borkdude.deps$_main.doInvoke(deps.clj:882)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at borkdude.deps.main(Unknown Source)
Jack-in process exited. Status: 1
```

## Workaround
Manually extracted all files from `clojure-tools.zip` into `ClojureTools` folder and replaced the existing files in the ClojureTools folder. Deleted the existing files, including `TRANSACTION_START`. After these steps, restarted VS Code and reran "Calva: Getting Started REPL," which resolved the issue.

## Environment Info
* Leiningen version: 2.11.2 on Java 1.8.0_341 Java HotSpot(TM) 64-Bit Server VM.
* Calva version in VS Code is the latest as of April 8, 2024.

**Comment by PEZ** [comment]
Hi! Thanks for reporting!

I think this happens upstream in deps.clj, so should probably also be reported there.

https://github.com/borkdude/deps.clj/blob/3f4bd66f5f6e75dc351140bab752ce4b6acf12f7/deps.clj#L928

WDYS, @borkdude?

**Comment by borkdude** [comment]
Feel free to post an issue with deps.clj WITH a repro

**Comment by borkdude** [comment]
I can't reproduce this with Windows + Java 8: 

https://github.com/borkdude/deps.clj/actions/runs/8649564947/job/23716093048

**Comment by borkdude** [comment]
but perhaps upgrading to Java 11 or so (or 21 if you're at it) might help

**Comment by sohang3112** [comment]
```console
$ javac -version
javac 17.0.1
```
@borkdude This is the Java version I'm using. Maybe upgrading Java will fix the issue - will let you know after I try it.
Also I'm using Windows 11 - don't know if that matters or not.

**Comment by borkdude** [comment]
> Leiningen version: 2.11.2 on Java 1.8.0_341 Java HotSpot(TM) 64-Bit Server VM.

This seems to indicate you are using java 1.8.0_341?

**Comment by sohang3112** [comment]
Just checked - apparently there were 2 Java versions installed on my system, and `lein` was using the old Java version. I changed `LEIN_JAVA_CMD` to `C:\Program Files\OpenJDK\jdk-17.0.1\bin\java.exe`, deleted `~/.deps.clj` and then did *Calva: Getting Started REPL* again - it worked now. Thanks 🙂

@borkdude Since the issue was due to an old Java version, can we add an early warning / error in case of old Java version?
